cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -0,0 +1,314 @@
+    appliance.db.client.session.query(rollups).delete()
+
+    # Chargeback reporting is done on hourly and daily rollup values and not real-time values.So, we
+    # are capturing C&U data and forcing hourly rollups by running these commands through
+    # the Rails console.
+
+    def verify_records_metrics_table(appliance, provider):
+        # Verify that rollups are present in the metric_rollups table.
+        vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+        ems = appliance.db.client['ext_management_systems']
+        metrics = appliance.db.client['metrics']
+
+        rc, out = appliance.ssh_client.run_rails_command(
+            "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+            vm.perf_capture('realtime', 2.hour.ago.utc, Time.now.utc)\""
+            .format(provider.id, repr(vm_name)))
+        assert rc == 0, "Failed to capture VM C&U data:".format(out)
+
+        with appliance.db.client.transaction:
+            result = (
+                appliance.db.client.session.query(metrics.id)
+                .join(ems, metrics.parent_ems_id == ems.id)
+                .filter(metrics.capture_interval_name == 'realtime',
+                metrics.resource_name == vm_name,
+                ems.name == provider.name, metrics.timestamp >= date.today())
+            )
+
+        for record in appliance.db.client.session.query(metrics).filter(
+                metrics.id.in_(result.subquery())):
+            if (record.cpu_usagemhz_rate_average or
+               record.cpu_usage_rate_average or
+               record.derived_memory_used or
+               record.net_usage_rate_average or
+               record.disk_usage_rate_average):
+                return True
+        return False
+
+    wait_for(verify_records_metrics_table, [appliance, provider], timeout=600,
+        fail_condition=False, message='Waiting for VM real-time data')
+
+    # New C&U data may sneak in since 1)C&U server roles are running and 2)collection for clusters
+    # and hosts is on.This would mess up our Chargeback calculations, so we are disabling C&U
+    # collection after data has been fetched for the last hour.
+
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+    rc, out = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_rollup_range(4.hour.ago.utc, Time.now.utc,'realtime')\"".
+        format(provider.id, repr(vm_name)))
+    assert rc == 0, "Failed to rollup VM C&U data:".format(out)
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=600, fail_condition=False,
+        message='Waiting for hourly rollups')
+
+    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
+    # in the metric_rollups DB table.The values from these hourly records are summed up.
+
+    with appliance.db.client.transaction:
```

Comment:
This looks like the same block of code that is in `verify_records_rollups_table` that triggers the true return. If so there is no need to run it again here, since wait_for didn't raise an exception you know this is query+subquery result is true and can set your average vars.

If I read it wrong and they aren't the same - apologies, ignore :)

Reply:
@nachandr and I discussed this offline, and while it is the same query, it has to be done again so that the records can be acted on. Maybe a future enhancement to make the verify_records_rollups_table return the records as well as the boolean.